straxen/plugins/online_monitor.py

```diff
@@ -228,50 +229,30 @@ def area_width_hist(self, data):
         return hist.T
 
 
-class OnlineVetoMonitor(strax.Plugin):
+class OnlineMonitor(strax.LoopPlugin):
     """
-    TODO
+    Loop over the online-monitor chunks, get the veto intervals that are within
+    each of these chunks. Compute the live-time within each of the chunks.
     """
-    depends_on = 'veto_intervals'
-    provides = 'online_veto_monitor'
-    __version__ = '0.0.1'
-    # data_kind = 'online_monitor'
+    depends_on = ('online_peak_monitor', 'veto_intervals')
+    provides = 'online_monitor'
+    __version__ = '0.0.4'
+    rechunk_on_save = False
 
     def infer_dtype(self):
-        dtype = [
-            (('Start time of the chunk', 'time'),
-             np.int64),
-            (('End time of the chunk', 'endtime'),
-             np.int64),
-            (('Live time', 'live_time'),
-             np.float64),
-
-        ]
+        dtype = strax.unpack_dtype(self.deps['online_peak_monitor'].dtype_for('online_peak_monitor'))
+        dtype += [(('Live time', 'live_time'),
+                   np.float64),]
         return dtype
 
-    def compute(self, veto_intervals, start, end):
-        # if not len(veto_intervals):
-        #     return np.empty(0, dtype=self.dtype)
-        res = np.ones(1, dtype=self.dtype)
-        res['time'] = start
-        res['endtime'] = end
-        if end-start > 0:
-            res['live_time'] = 1 - np.sum(veto_intervals['veto_interval']) / (end-start)
+    def compute_loop(self, peaks, veto_intervals):
+        res = {}
+        for d in peaks.dtype.names:
+            res[d] = peaks[d]
+        dt = strax.endtime(peaks) - peaks['time']
+        assert not np.iterable(dt) or len(dt) == 1
+        if dt > 0:
+            res['live_time'] = 1 - np.sum(veto_intervals['veto_interval'])/dt

```

Comment:
Just a small comment. For normal operation this is correct; however, in Veto TRG mode the np.sum(veto_intervals['veto_interval']) will actually be the live-time (i.e. veto off period). That's just how I made the start and stop signals in this mode. I don't think anything needs to be changed, just don't get surprised if you get the live-time reversed.

Reply:
Yeah, thanks Alex, you mentioned this. I thought it would make most sense to write it here like we expect it to be used. We can always change the interpretation on the analysts-side.

Reply:
Yeah I agree, hopefully we don't run with this Veto TRG mode for long :)